### PR TITLE
HIVE-26069: Remove unnecessary items from the .gitignore

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
Clean up gitignore, and also there is a config issue with some metastore tests which does not use the tmp dir

### Why are the changes needed?
When releasing the ignored files could remain in the src tarball, since they are not shown in the `git status` and `mvn clean` does not remove them

### Does this PR introduce _any_ user-facing change?
Nope

### How was this patch tested?
Tried to run several test.
```
mvn clean test -Dtest=TestBlobstoreNegativeCliDriver.java
mvn verify -DskipITests=false -Dit.test=ITestDerby -Dtest=nosuch
```
Also checked that `mvn datanucleus:enhance` does create the `datanucleus.log` file in the `target` dir

### CODE REVIEW

File: standalone-metastore/metastore-server/pom.xml [hunk 1]
```diff
@@ -573,6 +573,9 @@
             <java.io.tmpdir>${test.tmp.dir}</java.io.tmpdir>
             <test.tmp.dir>${test.tmp.dir}</test.tmp.dir>
             <hive.in.test>true</hive.in.test>
+            <derby.version>${derby.version}</derby.version>
+            <derby.stream.error.file>${test.tmp.dir}/derby.log</derby.stream.error.file>
+            <javax.jdo.option.ConnectionURL>jdbc:derby:memory:${test.tmp.dir}/junit_metastore_db;create=true</javax.jdo.option.ConnectionURL>
```
Comment (zabetak) [inline standalone-metastore/metastore-server/pom.xml:578]: This is a `MetastoreConf` property, it doesn't feel right to put it here. Also, I am not sure if every test should use an Derby database and specifically in-memory. Why is this change needed?

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public String getInitialJdbcUrl(String hostAddress) {
   }
 
   public String getDb() {
-    return MetaStoreServerUtils.JUNIT_DATABASE_PREFIX;
+    return "${test.tmp.dir}/" + MetaStoreServerUtils.JUNIT_DATABASE_PREFIX;
```
Comment (zabetak) [inline standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java:91]: I don't think this is necessary. First it is not going to be replaced like that and second it is used in conjunction with an in-memory derby so no DB file is created in the file-system.

Comment (zabetak) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java:91]: https://db.apache.org/derby/docs/10.8/devguide/cdevdvlp17453.html

File: standalone-metastore/metastore-server/pom.xml (hunk 1 above)
Comment (pvary) [reply standalone-metastore/metastore-server/pom.xml]: The following line is needed, and the derby.log file is generated inside the package directory for `ITestDerby` test.
```
<derby.stream.error.file>${test.tmp.dir}/derby.log</derby.stream.error.file>
```

Removed the `javax.jdo.option.ConnectionURL` property

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java (hunk 2 above)
Comment (pvary) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java]: We already use the prefix for the in memory datastore in multiple places:
https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/pom.xml#L395

https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/MetaStoreFactoryForTests.java#L90-L92

https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreServerUtils.java#L841-L842

I am not saying that this is good, I just state that we already use this several times already 😄.
Had to remove this change anyway, since the test failures for `TestMiniLlapLocalCliDriver`, `TestCliDriver`, ... are caused by this and the fix is not straightforward.

Comment (zabetak) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java]: I wasn't aware of those usages but for sure we can do much better than this. Since you reverted the change anyways we can discuss this separately.

**Review by zabetak — APPROVED** [review]
LGTM, pending tests!
